src/core/qgsdataprovider.h

```diff
@@ -66,6 +64,11 @@ class CORE_EXPORT QgsDataProvider : public QObject
 
   public:
 
+    /**
+     * Used in browser model to understand which items for which providers should be populated
+     *
+     * TODO: remove in QGIS 4 as this enum is really meant for data items rather than data providers!
```

Comment:
I don't think it exclusively is -- e.g. it's used by the bad layers handler dialog to identify file-based providers.

Reply:
As addressed above about bad layers...

Reply:
If the TODO remains, please move it to a normal (not doxygen) comment and replace it with a deprecated comment explaining the migration path.